subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GroovyScriptClassCompiler.java

```diff
@@ -241,6 +248,11 @@ public GroovyScriptCompilationAndInstrumentation(
             this.scriptCompilationHandler = scriptCompilationHandler;
         }
 
+        @Override
+        public Optional<CachingDisabledReason> shouldDisableCaching(@Nullable OverlappingOutputs detectedOverlappingOutputs) {
+            return Optional.of(NOT_CACHEABLE);
+        }
```

Comment:
:x: Let's add a comment about why we are disabling caching here (and why here and not for Kotlin DSL).